### PR TITLE
support overwriting dunder attributes of `BaseModel` instances

### LINKED ISSUE #3777: bug: cannot instantiate PEP 593 type alias
```python
from dataclasses import dataclass
from typing import Annotated
from pydantic import BaseModel

@dataclass
class DataclassModel:
    foo: str = "123"

DataclassAlias = Annotated[DataclassModel, "bar baz"]

DataclassAlias()  # works

class PydanticModel(BaseModel):
    foo: str = "123"

PydanticAlias = Annotated[PydanticModel, "bar baz"]

PydanticAlias()  # error
#   File ".../typing.py", line 680, in __call__
#     result.__orig_class__ = self
#   File "pydantic/main.py", line 347, in pydantic.main.BaseModel.__setattr__
# ValueError: "PydanticModel" object has no field "__orig_class__"
```

### PR DESCRIPTION
**Pull request by PrettyWood**
## Change Summary
After creating an instance of a `BaseModel`, we couldn't change its dunder attributes like `__annotations__`, `__doc__`, ...
This is rarely useful but `typing. Annotated` for example adds (i.e. sets) a `__orig_class__` attribute, which is forbidden with our implementation of `__setattr__`

## Related issue number
closes #3777

## Checklist
* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

**Comment by PrettyWood** [comment]
please review

File: tests/test_main.py [hunk 1]
```diff
@@ -2156,3 +2157,15 @@ class Model(BaseModel):
...
+def test_annotated_class():
+    class PydanticModel(BaseModel):
+        foo: str = '123'
+
+    PydanticAlias = Annotated[PydanticModel, 'bar baz']
+
+    pa = PydanticAlias()
+    pa.__doc__ = 'qwe'
+    assert repr(pa) == "PydanticModel(foo='123')"
+    assert pa.__doc__ == 'qwe'
```
Comment (adriangb) [inline tests/test_main.py:2171]: Test looks good to me 👍, it addresses #3777 as described

**Review by adriangb — APPROVED** [review]

**Review by samuelcolvin — APPROVED** [review]
LGTM, @PrettyWood should we include this in v1.9.1 or wait for v1.10?

I'm flexible but I think technically it doesn't count as a 1.9 fix. WDYT?

**Comment by PrettyWood** [comment]
I would add it in v1.10 and try to release it before this summer, couple of weeks after v1.9.1

File: tests/test_main.py [hunk 2]
```diff
@@ -2156,3 +2157,15 @@ class Model(BaseModel):
...
+
+
+def test_annotated_class():
+    class PydanticModel(BaseModel):
+        foo: str = '123'
+
+    PydanticAlias = Annotated[PydanticModel, 'bar baz']
+
+    pa = PydanticAlias()
+    pa.__doc__ = 'qwe'
```
Comment (adriangb) [inline tests/test_main.py:2169]:
```suggestion
    pa = PydanticAlias()
    assert isinstance(pa, PydanticModel)
```

I think it makes the test a bit clear, it's not immediately obvious what the result of instantiating something wrapped with `Annotated` should be.

Comment (samuelcolvin) [reply tests/test_main.py]: done.

**Comment by samuelcolvin** [comment]
thanks so much.

**Pull request merged**